requests/src/main/scala/weco/api/requests/services/SierraRequestsService.scala

```diff
@@ -182,6 +194,17 @@ class SierraRequestsService(
         None
     }
 
+  private def checkIfUserIsSelfRegistered(
+    patron: SierraPatronNumber
+  ): Future[Option[HoldRejected]] =
+    sierraSource
+      .lookupPatronType(patron)
+      .map {
+        case Right(Some(int)) if int == 29 =>
+          Some(HoldRejected.UserIsSelfRegistered)
+        case _ => None
+      }
```

Comment:
I think it might be better to remove the "opinion" of the type of `HoldRejected` from this method and make it - as its naming kind of encourages - return a `Future[Boolean]`, something like:

```scala
  private def checkIfUserIsSelfRegistered(
    patron: SierraPatronNumber
  ): Future[Boolean] =
    sierraSource
      .lookupPatronType(patron)
      .map {
        case Right(Some(29)) => true
        case _               => false
      }
```
in this case maybe the naming could be simplified, eg to `userIsSelfRegistered`